guides/quickstart.tt

```diff
@@ -0,0 +1,187 @@
+</pre>
+
+<p>(For security, you may want to <a href="[%root%]download.html#nix-verify-installation">verify the script</a> using our GPG signatures.)</p>
+
+<p>Check that the installation was successful by running</p>
+
+<pre class="code">
+nix-channel --list
+</pre>
+
+<p>This command displays the package distribution channel used by Nix. By default, this is <code>https://nixos.org/channels/nixpkgs-unstable</code></p>
+
+
+<h2>Your first shell</h2>
+
+<p>Nix can create reproducible environments given a declarative configuration called a Nix expression. Reproducible means you can share the configuration with others and guarantee that they are using the same software as you.</p>
+
+<p>To get started, make a new folder and create a file called <code>shell.nix</code> with the following contents:</p>
+
+<pre class="code">
+{ pkgs ? import &lt;nixpkgs&gt; {} }:
+
+pkgs.mkShell {
+  buildInputs = [
+    pkgs.which
+    pkgs.htop
+  ];
+}
+</pre>
+
+<p>Basically we import our package channel <code>nixpkgs</code> and make a shell with <code>which</code> and <code>htop</code> as inputs. To enter this environment, type in:</p>
+
+<pre class="code">
+nix-shell
+</pre>
+
+<p>The command will start downloading the missing packages from the cache. This may take a few moments. When it is done, you are dropped into a new shell. This shell provides the packages specified in <code>shell.nix</code>.</p>
+
+<p>Run <code>htop</code> to confirm it is present. Quit the program again by hitting Q.</p>
+
+<p>Now try <code>which htop</code> to check where the <code>htop</code> command is on-disk. You should see something similar to this:</p>
+
+<pre class="code">
+/nix/store/y3w2i8kfdbfj9rx287ad52rahjpgv423-htop-2.2.0/bin/htop
+</pre>
+
+<p>This is the path to the binary in the Nix store. Nix installs all packages into the store using a combination of its hash, name and version.</p>
+
+<p>You can search for available packages using <code>nix search</code>, for example:</p>
+
+<pre class="code">
+nix search python3
+nix search nodejs
+nix search ghc
+nix search cargo
+</pre>
+
+
+<h2>Setup a development environment</h2>
+
+<p>As an exercise, let us build a Python web application using the Flask web framework.</p>
+
+<p>Create a new file <code>default.nix</code>. This file is conventionally used for specifying packages:</p>
+
+<pre class="code">
+{ pkgs ? import &lt;nixpkgs&gt; {} }:
+
+pkgs.python3Packages.buildPythonApplication {
+  pname = "myapp";
+  src = ./.;
+  version = "0.1";
+  propagatedBuildInputs = [ pkgs.python3Packages.flask ];
+}
+</pre>
+
+<p>You will also need a simple Flask app as <code>main.py</code>:</p>
+
+<pre class="code">
+#! /usr/bin/env python
+
+from flask import Flask
+
+app = Flask(__name__)
+
+@app.route("/")
+def hello():
+    return "Hello, Nix!"
+
+if __name__ == "__main__":
+    app.run(host='0.0.0.0')
+</pre>
+
+<p>and a <code>setup.py</code> script:</p>
+
+<pre class="code">
+from distutils.core import setup
```

Comment:
`distutils` is best avoided. While it is still part of the stdlib its functionality has been superseded by `setuptools`.